parlai/core/utils.py

```diff
@@ -24,6 +24,7 @@
 
 """Near infinity, useful as a large penalty for scoring when inf is bad."""
 NEAR_INF = 1e20
+NEAR_INF_FP16 = 65504
```

Comment:
for some reason I find it funny that `NEAR_INF` for FP16 is 65k

Reply:
it's terrifying, no?

Reply:
Why do we use 65504 instead of 65535?

Reply:
Because it's not an uint16, it's a float16. That the limits are similar is a coincidence:

```
$ python -c 'import torch; print(torch.HalfTensor([65535]))'
tensor([inf], dtype=torch.float16)
$ python -c 'import torch; print(torch.HalfTensor([65504]))'
tensor([65504.], dtype=torch.float16)
$ python -c 'import torch; print(torch.HalfTensor([65505]))'
tensor([65504.], dtype=torch.float16)
```